sd-core/src/language/mod.rs

```diff
@@ -65,7 +66,7 @@ pub trait Language {
     type Addr: Syntax;
     type BlockAddr: Syntax;
     type VarDef: Syntax + GetVar<Self::Var>;
-    type Symbol: Syntax;
+    type Symbol: Syntax + TryFrom<Self::Addr, Error: Debug>;
```

Comment:
What change needs this TryFrom instance? It's not super clear to me what this does.

Reply:
It's so that we can generate a symbol from function names in LLVM IR.
Used here:
* https://github.com/sd-visualiser/sd-visualiser/pull/93/files#diff-b43459874fd946e94a7d7411c4bed0562fa1f026c2e50ab18eb1461f6471fbd5R219
* https://github.com/sd-visualiser/sd-visualiser/pull/93/files#diff-b43459874fd946e94a7d7411c4bed0562fa1f026c2e50ab18eb1461f6471fbd5R320
* https://github.com/sd-visualiser/sd-visualiser/pull/93/files#diff-2c586a5aa7319fca1c71e3013ffbff73f71d2dfcfa06601f48b47ef32dbcf6d3R101
Admittedly this way of doing it is kind of a hack - lifting `OpInfo` to cover more than just operations is an alternative (?)

Reply:
Ok, I think I get it now. I'd prefer for this to be a trait on `Addr` similar to `OpInfo` but it's fine as is if you want to leave it. As a side note `Addr` is an awful name, took me a while to remember what that part did. "ThunkName" or "ThunkId" would probably be better